### ISSUE TITLE
Re: CRAN package sf

### ISSUE THREAD

**Issue by edzer** [r-spatial/sf#1561]
@rsbivand 
Mail from CRAN:
```
> Dear maintainer,
> Please see the problems shown on
> <https://cran.r-project.org/web/checks/check_results_sf.html>.

> Please correct before 2021-01-06 to safely retain your package on
> CRAN.

This is about the Debian check errors, currently only for
r-devel-linux-x86_64-debian-gcc: presumably from a recent system upgrade
to GEOS 3.9 ...
```
With
```
> library(sf)
Linking to GEOS 3.9.0, GDAL 3.2.0, PROJ 7.2.0
```
I cannot reproduce the error, so probably https://github.com/r-spatial/sf/commit/c2216054a24f0a0a2851f1e26db0e8302740d596 (which was after current sf was released on CRAN) has solved it.

```

**Comment by edzer** [comment]
Looks like it, as with testing 0.9-6 (CRAN release) in the same container gives
```
  Running 'testthat.R'
 ERROR
Running the tests in 'tests/testthat.R' failed.
Last 13 lines of output:
  11/14 mismatches (average diff: 0.705)
  [1]  0.00 - 0.75 == -0.75
  [2]  1.00 - 0.00 ==  1.00
  [3]  2.00 - 1.00 ==  1.00
  [4]  1.25 - 2.00 == -0.75
  [5]  1.00 - 1.25 == -0.25
  [6]  0.75 - 1.00 == -0.25
  [7]  0.00 - 0.75 == -0.75
  [9]  2.50 - 1.25 ==  1.25
  [10] 1.25 - 2.50 == -1.25
  ...
  
  [ FAIL 2 | WARN 11 | SKIP 60 | PASS 733 ]
  Error: Test failures
  Execution halted
```

**Comment by rsbivand** [comment]
I agree, my current **sf** fork with released GEOS 3.9.0 passes check.  Later we may need to change the OverlayNG test to just checking the version, because I'm not sure that returned component order is deterministic, but I'm not sure yet.
